src/Policeman/Download/Hackage.hs

```diff
@@ -0,0 +1,53 @@
+{- | Functions to download Haskell packages from Hackage.
+-}
+
+module Policeman.Download.Hackage
+    ( PackageName (..)
+    , downloadFromHackage
+    ) where
+
+import Control.Exception (catch, throwIO)
+import Shellmet (($?))
+import System.Directory (createDirectoryIfMissing, removeDirectoryRecursive)
+import System.FilePath ((</>))
+import System.IO.Error (IOError, isDoesNotExistError)
+
+import Policeman.Core.Package (PackageName (..), Version (..))
+import Policeman.Download.Common (DownloadError (..), evidenceDir)
+
+
+{- | This function takes 'PackageName' and previous package
+'Version', downloads @.tar.gz@ archive from Hackage and unpacks it in
+the current directory.
+-}
+downloadFromHackage :: PackageName -> Version -> IO (Either DownloadError ())
+downloadFromHackage packageName@(PackageName name) (Version version) = do
+    let fullName = name <> "-" <> version
+    let tarName = fullName <> ".tar.gz"
+    let tarUrl = mconcat
+            [ "http://hackage.haskell.org/package/"
+            , name
+            , "/"
+            , tarName
+            ]
+
+    let tarPath = toText $ evidenceDir </> toString tarName
+    let srcPath = evidenceDir </> toString fullName
+    createDirectoryIfMissing True evidenceDir
+    removeDirIfExists srcPath
```

Comment:
Should we catch errors in here somehow?

Reply:
Oh now I see that you throw IO exception on `removeDirIfExists`, sorry!
But what do you think about making it `Left` our error instead?

Reply:
That's a nice idea!

Reply:
I've created an issue for refactoring:

* https://github.com/kowainik/policeman/issues/13